text/000-rustdoc-cfgs-handling.md

```diff
@@ -0,0 +1,283 @@
+pub mod futures {}
+```
+
+It will display in the documentation for this module:
+
+![This is supported on feature="futures-io" only.](https://user-images.githubusercontent.com/81079/89731116-d7b7ce00-da44-11ea-87c6-022d192d6eca.png)
+
+This attribute has the same syntax as conditional compilation, but it only causes documentation to be added. This means `#[doc(cfg(false))` will not cause your docs to be hidden, even though `#[cfg(false)]` does do that.
+
+This attribute works on modules and on items but cannot be used at the crate root level.
+
+### `#[doc(auto_cfg)]`/`#[doc(no_auto_cfg)]`
+
+By default, `#[doc(auto_cfg)]` is enabled at the crate-level. When it's enabled, Rustdoc will automatically display `cfg(...)` compatibility information as-if the same `#[doc(cfg(...))]` had been specified.
+
+So if we take back the previous example:
+
+```rust
+#[cfg(feature = "futures-io")]
+pub mod futures {}
+```
+
+There's no need to "duplicate" the `cfg` into a `doc(cfg())` to make Rustdoc display it.
+
+In some situations, the detailed conditional compilation rules used to implement the feature might not serve as good documentation (for example, the list of supported platforms might be very long, and it might be better to document them in one place). To turn it off, add the `#[doc(no_auto_cfg)]` attribute.
+
+Both `#[doc(auto_cfg)]` and `#[doc(no_auto_cfg)]` attributes impact all there descendants. You can then enable/disable them by using the opposite attribute on a given item. They can be used as follows:
+
+```rust
+// As an inner attribute, all this module's descendants will have this feature
+// enabled.
+#![doc(auto_cfg)]
+
+// As an outer attribute. So in this case, `foo` and all its
+// descendants won't have the `auto_cfg` feature enabled.
+#[doc(no_auto_cfg)] 
+pub mod foo {
+    // We re-enable the feature on `Bar` and on all its descendants.
+    #[doc(auto_cfg)]
+    pub struct Bar {
+        pub f: u32,
+    }
+}
+```
+
+As mentioned, both attributes can be used on modules, items and crate root level.
+
+### `#[doc(cfg_hide(...))]`
+
+This attribute is used to prevent some `cfg` to be generated in the visual markers. So in the previous example:
+
+```rust
+#[cfg(any(doc, feature = "futures-io"))]
+pub mod futures {}
+```
+
+It currently displays both `doc` and `feature = "futures-io"` into the documentation, which is not great. To prevent the `doc` cfg to ever be displayed, you can use this attribute at the crate root level:
+
+```rust
+#![doc(cfg_hide(doc))]
+```
+
+Or directly on a given item/module as it covers any of the item's descendants:
+
+```rust
+#[doc(cfg_hide(doc))]
+#[cfg(any(doc, feature = "futures-io"))]
+pub mod futures {
+    // `futures` and all its descendants won't display "doc" in their cfgs.
+}
+```
+
+Then, the `doc` cfg will never be displayed into the documentation.
+
+Rustdoc currently hides `doc` and `doctest` attributes by default and reserves the right to change the list of "hidden by default" attributes.
```

Comment:
It would be good to not use default-hidden predicates in the examples above then (another good common example could be `fuzzing`).

Reply:
👍